app/src/main/java/study_dev/testbed/studyhopper/CreateGroup.java

```diff
@@ -0,0 +1,283 @@
+        purple = findViewById(R.id.image_view_purple);
+        orange = findViewById(R.id.image_view_orange);
+        brown = findViewById(R.id.image_view_brown);
+        gray = findViewById(R.id.image_view_gray);
+
+        blue.setOnClickListener(this);
+        green.setOnClickListener(this);
+        yellow.setOnClickListener(this);
+        red.setOnClickListener(this);
+        purple.setOnClickListener(this);
+        orange.setOnClickListener(this);
+        brown.setOnClickListener(this);
+        gray.setOnClickListener(this);
+
+        getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_close);
+        setTitle("Create Group");
+
+        // Enable back button
+        ActionBar supportActionBar = getSupportActionBar();
+        if (supportActionBar != null) {
+            supportActionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        groupPreferencesSpinner = findViewById(R.id.spinner_group_preferences);
+        fillGroupPreferencesSpinner();
+    }
+
+    public void createStudyGroup(View v) {
+        String groupName = editTextGroupName.getText().toString();
+        String courseCode = editTextCourseCode.getText().toString();
+        String groupPreference = groupPreferencesSpinner.getSelectedItem().toString();
+        int groupMaxSize = Integer.parseInt(editTextMaxSize.getText().toString());
+        boolean coedGroup = false, femalesOnlyGroup = false, malesOnlyGroup = false;
+
+        if(groupName.trim().isEmpty() || courseCode.trim().isEmpty()) {
+            Toast.makeText(this, "Please input a group name and course code!", Toast.LENGTH_SHORT).show();
+            return;
+        }
+            
+        if(groupPreference.trim().isEmpty()) {
+            Toast.makeText(this, "Please select a group preference from the dropdown!", Toast.LENGTH_SHORT).show();
+            return;
+        }
+
+        if(groupMaxSize == 0) {
+            Toast.makeText(this, "Please select the maximum number of members for the group!", Toast.LENGTH_SHORT).show();
+            return;
+        }
+
+        if (colorSelected == 0)
+        {
+            Toast.makeText(this, "Please select a color for the group", Toast.LENGTH_SHORT).show();
+            return;
+        }
+
+        if(groupPreference.equals("Coed Group"))
+            coedGroup = true;
+        else if(groupPreference.equals("Females Only Group"))
+            femalesOnlyGroup = true;
+        else if(groupPreference.equals("Males Only Group"))
+            malesOnlyGroup = true;
+
+        // Firebase reference for "Groups" Collection
+        CollectionReference groupRef = FirebaseFirestore.getInstance()
+                .collection("groups");
+        // Firebase Reference for "Groups" in users sub-collection
+        CollectionReference userGroupRef = FirebaseFirestore.getInstance()
+                .collection("users").document(getUserName())
+                .collection("groups");
+
+        groupRef.add(new Group(groupName, courseCode, groupColor,
```

Comment:
if using the same group, it would be better to save it as a variable and pass it to both dbs instead of declaring a new instance each time

Reply:
Passed a single Group instance intro groupRef and userGroupRef.